node/add_validator_primary.go

```diff
@@ -0,0 +1,195 @@
+) (ids.ID, error) {
+	if validator.NodeID == ids.EmptyNodeID {
+		return ids.Empty, subnet.ErrEmptyValidatorNodeID
+	}
+
+	if validator.Duration == 0 {
+		return ids.Empty, subnet.ErrEmptyValidatorDuration
+	}
+
+	minValStake, err := GetMinStakingAmount(network)
+	if err != nil {
+		return ids.Empty, err
+	}
+
+	if validator.StakeAmount < minValStake {
+		return ids.Empty, fmt.Errorf("invalid weight, must be greater than or equal to %d: %d", minValStake, validator.StakeAmount)
+	}
+
+	if validator.DelegationFee == 0 {
+		validator.DelegationFee = network.GenesisParams().MinDelegationFee
+	}
+
+	if err = h.HandleBLSKey(); err != nil {
+		return ids.Empty, fmt.Errorf("unable to set BLS key of node due to %w", err)
+	}
+
+	wallet.SetSubnetAuthMultisig([]ids.ShortID{})
+
+	owner := &secp256k1fx.OutputOwners{
+		Threshold: 1,
+		Addrs: []ids.ShortID{
+			wallet.Addresses()[0],
+		},
+	}
+
+	proofOfPossession := signer.NewProofOfPossession(h.BlsSecretKey)
+	nodeID, err := ids.NodeIDFromString(h.NodeID)
+	if err != nil {
+		return ids.Empty, err
+	}
+
+	unsignedTx, err := wallet.P().Builder().NewAddPermissionlessValidatorTx(
+		&txs.SubnetValidator{
+			Validator: txs.Validator{
+				NodeID: nodeID,
+				End:    uint64(time.Now().Add(validator.Duration).Unix()),
+				Wght:   validator.StakeAmount,
+			},
+			Subnet: ids.Empty,
+		},
+		proofOfPossession,
+		wallet.P().Builder().Context().AVAXAssetID,
+		owner,
+		owner,
+		validator.DelegationFee,
+	)
+	if err != nil {
+		return ids.Empty, fmt.Errorf("error building tx: %w", err)
+	}
+
+	tx := txs.Tx{Unsigned: unsignedTx}
+	if err := wallet.P().Signer().Sign(context.Background(), &tx); err != nil {
+		return ids.Empty, fmt.Errorf("error signing tx: %w", err)
+	}
+
+	ctx, cancel := utils.GetAPIContext()
+	defer cancel()
+	err = wallet.P().IssueTx(
+		&tx,
+		common.WithContext(ctx),
+	)
+	if err != nil {
+		if ctx.Err() != nil {
+			err = fmt.Errorf("timeout issuing/verifying tx with ID %s: %w", tx.ID(), err)
+		} else {
+			err = fmt.Errorf("error issuing tx with ID %s: %w", tx.ID(), err)
+		}
+		return ids.Empty, err
+	}
+
+	return tx.ID(), nil
+}
+
+func GetMinStakingAmount(network avalanche.Network) (uint64, error) {
+	pClient := platformvm.NewClient(network.Endpoint)
+	ctx, cancel := utils.GetAPIContext()
+	defer cancel()
+	minValStake, _, err := pClient.GetMinStake(ctx, ids.Empty)
+	if err != nil {
+		return 0, err
+	}
+	return minValStake, nil
+}
+
+func (h *Node) SetNodeBLSKey(signingKeyPath string) error {
+	blsKeyBytes, err := os.ReadFile(signingKeyPath)
+	if err != nil {
+		return err
+	}
+	blsSk, err := bls.SecretKeyFromBytes(blsKeyBytes)
+	if err != nil {
+		return err
+	}
+	h.BlsSecretKey = blsSk
+	return nil
+}
+
+func RemoveTmpSDKDir() error {
+	usr, err := user.Current()
+	if err != nil {
+		return fmt.Errorf("unable to get system user %w", err)
+	}
+	return os.RemoveAll(filepath.Join(usr.HomeDir, constants.LocalTmpDir))
+}
+
+func (h *Node) GetBLSKeyFromRemoteHost() error {
```

Comment:
this func should return blskey []bytes so it can be handled by user what to do with it